File: binding/src/opengl/libqopenglversionprofile.cpp
#include <QOpenGLVersionProfile>
#include <QSurfaceFormat>
#include <qopenglversionprofile.h>
#include "libqopenglversionprofile.h"
#include "libqopenglversionprofile.hxx"

QOpenGLVersionProfile* QOpenGLVersionProfile_new() {
	 return new QOpenGLVersionProfile();
}

QOpenGLVersionProfile* QOpenGLVersionProfile_new2(const QSurfaceFormat* format) {
	 return new QOpenGLVersionProfile(*format);
}

QOpenGLVersionProfile* QOpenGLVersionProfile_new3(const QOpenGLVersionProfile* other) {
	 return new QOpenGLVersionProfile(*other);
}

void QOpenGLVersionProfile_OperatorAssign(QOpenGLVersionProfile* self, const QOpenGLVersionProfile* rhs) {
	self->operator=(*rhs);
}

libqt_pair QOpenGLVersionProfile_Version(const QOpenGLVersionProfile* self) {
	return self->version();
}

void QOpenGLVersionProfile_SetVersion(QOpenGLVersionProfile* self, int majorVersion, int minorVersion) {
	self->setVersion(majorVersion, minorVersion);
}

int QOpenGLVersionProfile_Profile(const QOpenGLVersionProfile* self) {
	return self->profile();
}

void QOpenGLVersionProfile_SetProfile(QOpenGLVersionProfile* self, int profile) {
	self->setProfile(static_cast<QSurfaceFormat::OpenGLContextProfile>(profile));
}

bool QOpenGLVersionProfile_HasProfiles(const QOpenGLVersionProfile* self) {
	return self->hasProfiles();
}

bool QOpenGLVersionProfile_IsLegacyVersion(const QOpenGLVersionProfile* self) {
	return self->isLegacyVersion();
}

bool QOpenGLVersionProfile_IsValid(const QOpenGLVersionProfile* self) {
	return self->isValid();
}

void QOpenGLVersionProfile_Delete(QOpenGLVersionProfile* self) {
    delete self;
}

File: binding/include/opengl/libqopenglversionprofile.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qopenglversionprofile.html
pub const qopenglversionprofile = struct {

    /// New constructs a new QOpenGLVersionProfile object.
    pub fn New() QtC.QOpenGLVersionProfile {
        return qtc.QOpenGLVersionProfile_new();
    }


    /// New2 constructs a new QOpenGLVersionProfile object.
    pub fn New2(format: ?*anyopaque) QtC.QOpenGLVersionProfile {
        return qtc.QOpenGLVersionProfile_new2(@ptrCast(format));
    }


    /// New3 constructs a new QOpenGLVersionProfile object.
    pub fn New3(other: ?*anyopaque) QtC.QOpenGLVersionProfile {
        return qtc.QOpenGLVersionProfile_new3(@ptrCast(other));
    }


    pub fn OperatorAssign(self: ?*anyopaque, rhs: ?*anyopaque) void {
        qtc.QOpenGLVersionProfile_OperatorAssign(@ptrCast(self), @ptrCast(rhs));
    }

    pub fn Version(self: ?*anyopaque, ) QPair<int, int> {
        return qtc.QOpenGLVersionProfile_Version(@ptrCast(self));
    }

    pub fn SetVersion(self: ?*anyopaque, majorVersion: i32, minorVersion: i32) void {
        qtc.QOpenGLVersionProfile_SetVersion(@ptrCast(self), majorVersion, minorVersion);
    }

    pub fn Profile(self: ?*anyopaque, ) i32 {
        return qtc.QOpenGLVersionProfile_Profile(@ptrCast(self));
    }

    pub fn SetProfile(self: ?*anyopaque, profile: i32) void {
        qtc.QOpenGLVersionProfile_SetProfile(@ptrCast(self), @intCast(profile));
    }

    pub fn HasProfiles(self: ?*anyopaque, ) bool {
        return qtc.QOpenGLVersionProfile_HasProfiles(@ptrCast(self));
    }

    pub fn IsLegacyVersion(self: ?*anyopaque, ) bool {
        return qtc.QOpenGLVersionProfile_IsLegacyVersion(@ptrCast(self));
    }

    pub fn IsValid(self: ?*anyopaque, ) bool {
        return qtc.QOpenGLVersionProfile_IsValid(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QOpenGLVersionProfile_Delete(@ptrCast(self));
    }
};
